validator/accounts/v2/new.go

```diff
@@ -42,43 +44,44 @@ func NewAccount(cliCtx *cli.Context) error {
 		log.Fatal(err)
 	}
 	ctx := context.Background()
-	keymanagerKind, err := readKeymanagerKindFromWalletPath(walletDir)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	// Only direct keymanagers can create accounts for now.
-	if keymanagerKind == v2keymanager.Remote {
-		log.Fatal("Cannot create a new account for a remote keymanager")
-	}
-	// Read the directory for password storage from user input.
-	passwordsDirPath := inputPasswordsDirectory(cliCtx)
-	wallet, err := OpenWallet(ctx, &WalletConfig{
-		PasswordsDir:      passwordsDirPath,
+	wallet, err := OpenWallet(cliCtx, &WalletConfig{
```

Comment:
Simplified a lot of logic by making OpenWallet smarter, being able to detect the type of keymanager, ask for a wallet password if none specified, and more